src/vs/workbench/browser/parts/views/views.ts

```diff
@@ -796,41 +859,106 @@ export class ViewDescriptorService extends Disposable implements IViewDescriptor
 			return;
 		}
 
-		const from = this.viewContainers.get(views[0].id);
+		const from = this.getViewContainer(views[0].id);
 		const to = viewContainer;
 
 		if (from && to && from !== to) {
-			this.onDidDeregisterViews(from, views);
-			this.onDidRegisterViews(viewContainer, views);
-			this._onDidChangeContainer.fire({ views, from, to });
+			this.removeViews(from, views);
+			this.addViews(to, views);
+			this.saveViewPositionsToCache();
 		}
 	}
 
-	private onDidRegisterViewContainer(viewContainer: ViewContainer): void {
-		const disposables = new DisposableStore();
-		const viewDescriptorCollection = disposables.add(new ViewDescriptorCollection(this.contextKeyService));
-		viewDescriptorCollection.addViews(this.viewsRegistry.getViews(viewContainer));
+	private getViewContainerById(id: string): ViewContainer | undefined {
```

Comment:
Is this method does not exist on view container registry? If does not, lets please create one there and use it.